modules/log4j-configurator/src/main/java/org/mule/runtime/module/log4j/internal/ApplicationReconfigurableLoggerContextSelector.java

```diff
@@ -0,0 +1,109 @@
+    // The container logger context is created with no logging separation.
+    // This will guarantee that by default this will work as no separation in logs.
+    this.containerLoggerContext =
+        this.loggerContextFactory.build(SYSTEM_CLASSLOADER, this, false);
+  }
+
+  public LoggerContext getContext(String fqcn, ClassLoader loader, boolean currentContext) {
+    return containerLoggerContext;
+  }
+
+  public LoggerContext getContext(String fqcn, ClassLoader loader, boolean currentContext, URI configLocation) {
+    return getContext(fqcn, loader, currentContext);
+  }
+
+  public List<LoggerContext> getLoggerContexts() {
+    return asList(containerLoggerContext);
+  }
+
+  public void removeContext(LoggerContext context) {}
+
+  private static ClassLoader getLoggerClassLoader(ClassLoader loggerClassLoader) {
+    if (loggerClassLoader instanceof CompositeClassLoader) {
+      return getLoggerClassLoader(findClassLoader((CompositeClassLoader) loggerClassLoader));
+    }
+
+    // Obtains the first artifact class loader in the hierarchy
+    while (!(loggerClassLoader instanceof ArtifactClassLoader) && loggerClassLoader != null) {
+      loggerClassLoader = loggerClassLoader.getParent();
+    }
+
+    if (loggerClassLoader == null) {
+      loggerClassLoader = SYSTEM_CLASSLOADER;
+    } else if (isRegionClassLoaderMember(loggerClassLoader)) {
+      loggerClassLoader =
+          isPolicyClassLoader(loggerClassLoader.getParent()) ? loggerClassLoader.getParent().getParent()
+              : loggerClassLoader.getParent();
+    } else if (!(loggerClassLoader instanceof RegionClassLoader)
+        && !(loggerClassLoader instanceof MuleSharedDomainClassLoader)) {
+      loggerClassLoader = SYSTEM_CLASSLOADER;
+    }
+    return loggerClassLoader;
+  }
+
+  private static boolean isPolicyClassLoader(ClassLoader loggerClassLoader) {
+    return ((ArtifactClassLoader) loggerClassLoader).getArtifactDescriptor() instanceof PolicyTemplateDescriptor;
+  }
+
+  private static boolean isRegionClassLoaderMember(ClassLoader classLoader) {
+    return !(classLoader instanceof RegionClassLoader) && classLoader.getParent() instanceof RegionClassLoader;
+  }
+
+  public void reconfigureAccordingToAppClassloader(ClassLoader classloader) {
+    LoggerContext applicationClassLoaderLoggerContext = this.loggerContextFactory
+        .build(classloader, this, true);
+
+    applicationClassLoaderLoggerContext.reconfigure();
+
+    // We reconfigure the loggers that were already provided to get the app log4j configuration.
+    containerLoggerContext.updateLoggers(applicationClassLoaderLoggerContext.getConfiguration());
+
+    // We change the configuration for the logger context. This only sets the configuration.
+    containerLoggerContext.reconfigure(applicationClassLoaderLoggerContext.getConfiguration());
+
+    // This is needed so that the configuration set is reconfigured.
+    containerLoggerContext.reconfigure();
```

Comment:
what is the purpose of doing this on a context that is discarded at the end of the method?